### PR TITLE
Smile detection timer implementation

### PR DESCRIPTION
**Pull request by nemotime**
Added a requirement that the smile must be held for 2 seconds before a photo is taken, if the smile is interrupted or the camera is moved drastically, the countdown will reset automatically. This allows the user some time to get adjusted and shows a progress bar at the top of the page which increments in quarters. When the bar is full, that means that the photo has been taken after which the bar empties, ready for another photo.

Calling for reviewers: I was using a very old phone with an odd aspect ratio to test my addition. Please verify that the bar appears properly and the camera is not cut off drastically. 

As per issue #78

### CODE REVIEW

**Comment by n2chao** [comment]
The progress bar works on the pixel 2 but can you add a small animation or any can of feedback when the picture is taken? If there's no sound there's no way to tell if it triggered or not.

**Comment by nemotime** [comment]
@n2chao when the bar is full it takes a picture, I'll see if I can add some feedback. I tried changing the color of the bar from yellow to green but that requires API 21 and over.

**Comment by n2chao** [comment]
API 21 and over should be fine since it covers most phones from the past 5 years

File: app/src/main/kotlin/com/simplemobiletools/camera/activities/SmileDetectionActivity.kt [hunk 1]
```diff
@@ -29,7 +29,12 @@ import java.io.File
 import java.io.FileOutputStream
 import java.io.IOException
 import android.content.Intent
+import android.content.res.ColorStateList
```
Comment (Earthii) [inline app/src/main/kotlin/com/simplemobiletools/camera/activities/SmileDetectionActivity.kt]: Doesn't look like you used these imports. Clean up the imports please

File: app/src/main/kotlin/com/simplemobiletools/camera/activities/SmileDetectionActivity.kt [hunk 2]
```diff
@@ -29,7 +29,12 @@ import java.io.File
...
+import android.graphics.Color
```
Comment (Earthii) [inline app/src/main/kotlin/com/simplemobiletools/camera/activities/SmileDetectionActivity.kt]: Same

**Comment by Earthii** [comment]
I left comments on the unused imports.

Also, Is there a reason in particular why we needed a thread?

File: app/src/main/kotlin/com/simplemobiletools/camera/activities/SmileDetectionActivity.kt (hunk 2 above)
Comment (nemotime) [reply app/src/main/kotlin/com/simplemobiletools/camera/activities/SmileDetectionActivity.kt]: I used this in my latest commits

File: app/src/main/kotlin/com/simplemobiletools/camera/activities/SmileDetectionActivity.kt (hunk 1 above)
Comment (nemotime) [reply app/src/main/kotlin/com/simplemobiletools/camera/activities/SmileDetectionActivity.kt]: same

**Comment by nemotime** [comment]
@Earthii I could not get the progress bar to update properly without using a thread. It may be because I'm running on an S3 Mini or maybe I had errors in my code but the thread.sleep() method allows me to update the progress bar every 500millis while sleeping the thread in between incrementations

**Pull request merged**